File: 7.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x >= 0:
            sign = 1
        else:
            sign = -1
        rev = str(abs(x))[::-1]
        res = int(rev)*sign
        if res <-2**31 or res > 2**31-1:
            return 0
        return res
    

File: 7.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int temp = x % 10;
            x /= 10;
            if (res > Integer.MAX_VALUE/10 ||
                (res == Integer.MAX_VALUE/10 && temp > 7)) return 0;
            if (res < Integer.MIN_VALUE/10 ||
                (res ==Integer.MIN_VALUE/10 && temp < -8)) return 0;
            res = res*10+temp;
        }
        return res;
    }
}